packages/api/src/platforms/vtex/resolvers/validateCart.ts

```diff
@@ -413,12 +420,14 @@ export const validateCart = async (
     // update orderForm etag so we know last time we touched this orderForm
     .then((form) => setOrderFormEtag(form, commerce))
     .then(joinItems)
+  
+  // Step5: Update ShippingData from orderForm
+  await updateOrderFormShippingData(updatedOrderForm, session, ctx)
```

Comment:
We're changing the orderform after setting it's etag, which will lead to the etag being outdated, right? If so, this will cause the orderform to always be updated even when there was no external changes, which is not ideal. Does the process of updating the shipping data changes anything inside the items array? If so, the answer is yes.

This process should be done before generating the etag, and the orderform used to generate the etag should be the updated one.

Reply:
Good catch! I think in some cases the shipping can change the items array when we have the selected seller changing